mypy/subtypes.py

```diff
@@ -186,7 +186,10 @@ def visit_tuple_type(self, left: TupleType) -> bool:
                   is_named_instance(right, 'typing.Container') or
                   is_named_instance(right, 'typing.Sequence') or
                   is_named_instance(right, 'typing.Reversible')):
-                iter_type = right.args[0]
+                if right.args:
```

Comment:
There seems to be code that does just that in [typeanal.py at line 418](https://github.com/python/mypy/blob/master/mypy/typeanal.py#L418). So apparently it doesn't work or runs too late to help here?